### DISCUSSION TITLE
[Feature] Browse external libraries by their actual subdirectories

### DISCUSSION THREAD

**Discussion by TheAnachronism** [immich-app/immich#5110] [Feature Request]
### The feature

I know it was already quite the stretch to get external libraries into Immich, but what would improve it even further would be the functionality to browse the images by their actual subdirectory structure.
Right now, everything is placed upon the creation timeline, ignoring the existing directory structure.

It would be really awesome if an external library could actually browse like that.

I guess this would be quite the larger implementation, as “folder” structure in the web-panel (and maybe even on assets?) isn't considered right now.

This is probably related to topics like _nested albums_, etc.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by klaus1k** [comment]
I personally would prefer to unify the folders as albums. E.g. Photoprism has a separate albums and folder view and I always found it kind of confusing.
For me or would be perfect to have that functionality as stated in #2073 nested albums.
I think importing albums for libraries is already planned? (as it was in the classic CLI)

**Comment by TheAnachronism** [comment]
Yeah, I guess that works as well.
The important part for me would be, that I have some way of using the already organized structure of my external photos.

**Comment by etnoy** [comment]
This is on my personal to-do list, but I'm not sure how to architect this just yet

**Reply by haldi4803** [reply]
I also think that's a must have Feature for existing libraries.
BUT instead of creating an additional view for existing folder structure wouldn't it be way simpler and also achieve almost the same by simply adding all Fotos from Subfolder to an Album with the same Name?

/existingLibrary/Fotos
With subfolders Holiday1, holiday2, holiday3
Then creates Album Holiday1, with all Fotos from Subfolder Holiday1

If you have SubSubfolder /existingLibrary/Fotos/Holiday1/Nighfotos you could create a separate album called Holiday1 - Nightfotos.


Sounds more like a quick an dirty fix instead of a proper solution right?
And in case someone has a Date Folder Structure like
/existingLibrary/Fotos/Holiday1/2022-12-30 
You would get an Album for every day which is kinday annoying.
So as a Work around you'd need either an "ignore Date timed folders" or "ignore SubSubfolders" work around.


But I'm still in favour of 'properly' importing existing libraries with somehow using existing folder structure.


Edit: oh I see.
There is already a separate discussion about that. #4279

**Comment by adamlove86** [comment]
If we can already search by file name, how difficult would it be to just add the option (or change it) so it searches by path?

**Comment by bo0tzz** [comment]
Folder view support has now been implemented
